### PR TITLE
Fix more regressions caused by repo replacement for openSUSE

### PR DESCRIPTION
**Pull request by rwx788**
## Description
Run pkcon_quit before adding repos and do not schedule repo replacement if SUSE_MIRROR is not defined

In some scenarios we can have packagekit running which will make adding
repos impossible. So proactively disable it using consoletest_setup.
  
For Leap live images we don't have mirrors available, so should not try
to replace repos. But still need to do that for TW live images.

See [poo#32038](https://progress.opensuse.org/issues/32038).

After this is merged, we can remove KEEP_ONLINE_REPOS variable from all live leap installations and create_hdd_gnome.

### Verification runs:
 * [create_hdd_gnome](http://gershwin.arch.suse.de/tests/248#)
 * [leap live image](http://gershwin.arch.suse.de/tests/240)

### CODE REVIEW

File: lib/main_common.pm [hunk 1]
```diff
@@ -1696,7 +1697,11 @@ sub load_create_hdd_tests {
     loadtest 'console/hostname'       unless is_bridged_networking;
     loadtest 'console/force_cron_run' unless is_jeos;
     # Remove repos pointing to download.opensuse.org and add snaphot repo from o3
-    replace_opensuse_repos_tests if is_repo_replacement_required;
+    if (is_repo_replacement_required) {
+        loadtest 'console/consoletest_setup';
+        replace_opensuse_repos_tests;
+        loadtest 'console/consoletest_finish';
```
Comment (okurz) [inline lib/main_common.pm]: It was not the idea of "consoletest_setup" and "consoletest_finish" to wrap just one module. I doubt this is a good idea. Please just include the necessary parts from consoletest_setup within `replace_opensuse_repos_tests`. Certainly most parts are *not* necessary. And especially consoletest_finish I think should not be needed at all here.

Comment (rwx788) [reply lib/main_common.pm]: consoletest_finish reverts some parts we do in consoletest_setup. Like unmasking packagekit service. I'll check what we can do better here.

File: lib/main_common.pm [hunk 2]
```diff
@@ -1696,7 +1701,9 @@ sub load_create_hdd_tests {
     loadtest 'console/hostname'       unless is_bridged_networking;
     loadtest 'console/force_cron_run' unless is_jeos;
     # Remove repos pointing to download.opensuse.org and add snaphot repo from o3
-    replace_opensuse_repos_tests if is_repo_replacement_required;
+    if (is_repo_replacement_required) {
+        replace_opensuse_repos_tests;
+    }
```
Comment (okurz) [inline lib/main_common.pm]: Why did you change that? The inline is more common and I would like to keep the old approach. If you would change this back then LGTM

File: tests/update/zypper_clear_repos.pm [hunk 3]
```diff
@@ -18,6 +18,8 @@ use utils;
 
 sub run {
     select_console 'root-console';
+    # packagekit service may block zypper when operate on repos
+    pkcon_quit;
```
Comment (okurz) [inline tests/update/zypper_clear_repos.pm]: better approach
